### PR TITLE
Plugin/resolution

### PR DESCRIPTION
**Pull request by boi123212321**

### CODE REVIEW

File: plugins/resolution/main.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  };
+}
+
+module.exports = ({ args, $throw, data, scenePath, scene }: SceneContext & MyContext) => {
+  if (!scenePath) {
+    return $throw("Uh oh. You shouldn't use the plugin for this type of event");
+  }
+
+  let resolutions = [144, 240, 360, 480, 720, 1080, 2160];
+  if (args.resolutions) {
```
Comment (leadwolf) [inline plugins/resolution/main.ts:15]: it's possible for users to not even have `args`

File: plugins/resolution/info.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    "sceneCreated",
+    "sceneCustom"
+  ],
+  "arguments": [
+    {
+      "name": "resolutions",
+      "type": "number[]",
+      "required": false,
+      "default": [],
+      "description": "Resolutions to match with"
```
Comment (leadwolf) [inline plugins/resolution/info.json:16]:
```suggestion
      "description": "Resolutions to match with, against the scene's path, when the scene's metadata has not yet been extracted"
```

**Comment by leadwolf** [comment]
I know it's super basic, but there should be a fixture making sure that the `data` labels are merged.
Also, it would be nice to make small `docs.md` explaining that
- the plugin first looks at the scene's meta, and will always return that, if it exists
- `args.dimensions` is only used when the scene's meta doesn't exist. Show the default dimensions

File: plugins/resolution/main.ts (hunk 1 above)
Comment (boi123212321) [reply plugins/resolution/main.ts:15]: args always default to { }

Comment (leadwolf) [reply plugins/resolution/main.ts:15]: forgot that the plugin runner does that 😬

**Review by leadwolf — APPROVED** [review]

**Pull request merged**